libs/wingsdk/src/utils/resource-names.ts

```diff
@@ -0,0 +1,53 @@
+import { Construct } from "constructs";
+
+export enum CaseConventions {
+  LOWERCASE = "lowercase",
+  UPPERCASE = "uppercase",
+}
+
+export interface NameOptions {
+  /**
+   * Maximum name length must always be greater than the hash (>=8 characters)
+   */
+  readonly maxLen: number;
+  /**
+   * Regular expression that indicates which characters are valid
+   */
+  readonly allowedRegEx: RegExp;
```

Comment:
Can we make these fields optional? I think having no max length and allowing any characters makes sense as defaults. (be sure to mention defaults in the docstrings)

Reply:
I can change to Optional but I don't know if some cloud provider don't have some type of limitations on name resources

I will make this Optional

Reply:
Alright, let's start with this / adjust if there's a use case 👍